packages/ra-ui-materialui/src/input/AutocompleteInput.spec.tsx

```diff
@@ -1259,6 +1259,67 @@ describe('<AutocompleteInput />', () => {
             expect(screen.queryByText('ra.action.create')).toBeNull();
             expect(screen.queryByText('ra.action.create_item')).toBeNull();
         });
+        it('should not show the create option when a choice is selected when using a custom createLabel', async () => {
+            const choices = [
+                { id: 'ang', name: 'Angular' },
+                { id: 'rea', name: 'React' },
+            ];
+            const handleCreate = filter => {
+                const newChoice = {
+                    id: 'js_fatigue',
+                    name: filter,
+                };
+                choices.push(newChoice);
+                return newChoice;
+            };
+
+            render(
+                <AdminContext dataProvider={testDataProvider()}>
+                    <ResourceContextProvider value="posts">
+                        <SimpleForm
+                            mode="onBlur"
+                            onSubmit={jest.fn()}
+                            defaultValues={{ language: 'ang' }}
+                        >
+                            <AutocompleteInput
+                                source="language"
+                                choices={choices}
+                                onCreate={handleCreate}
+                                createLabel="Start typing to create a new item"
+                            />
+                        </SimpleForm>
+                    </ResourceContextProvider>
+                </AdminContext>
```

Comment:
Why don't you use the story?

Reply:
I copied the test just above which doesn't use it either, saving some time as it does almost exactly what I need in my own test.

Reply:
Done anyway, for both tests